lib/grammers-client/src/client/files.rs

```diff
@@ -152,6 +153,105 @@ impl ClientHandle {
         Ok(())
     }
 
+    /// Uploads an async buffer to Telegram servers.
+    ///
+    /// The file is not sent to any chat, but can be used as media when sending messages for a
+    /// certain period of time (less than a day). You can use this uploaded file multiple times.
+    ///
+    /// Refer to [`InputMessage`] to learn more uses for `uploaded_file`.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # async fn f(chat: grammers_tl_types::enums::InputPeer, mut client: grammers_client::ClientHandle, stream: &mut std::io::Cursor<Vec<u8>>) -> Result<(), Box<dyn std::error::Error>> {
+    /// use grammers_client::InputMessage;
+    ///
+    /// let uploaded_file = client.upload_stream(stream, Some("sleep.jpg".to_string())).await?;
+    ///
+    /// client.send_message(&chat, InputMessage::text("Check this out!").photo(uploaded_file)).await?;
+    /// # Ok(())
+    /// # }
+    /// ```
+    ///
+    /// [`InputMessage`]: crate::InputMessage
+
+    pub async fn upload_stream<S: AsyncRead + AsyncSeek + Unpin>(
+        // Do we want Sized instead of AsyncSeek?
```

Comment:
`Sized` won't tell you the size of the file though? I don't see the purpose of this comment.

I think the right approach here is to take the file size as input parameter, and read that much from the stream. Think HTTP streams, may not be seekable, but you might know the file size beforehand.